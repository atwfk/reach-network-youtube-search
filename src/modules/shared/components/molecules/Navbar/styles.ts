import styled from "styled-components";
import { SCREENS } from "../../../constants";

export const StyledNav = styled.nav`
  background-color: red;
  box-shadow: 0px 4px 4px #f1f2f3;
  padding: 5px 20px;

  @media (min-width: ${SCREENS.MD}) {
    background-color: white;
    padding: 10px 20px;
  }

  .nav-container {
    max-width: ${SCREENS.LG};
    margin: auto;
    display: flex;
    justify-content: space-between;
  }
`;
