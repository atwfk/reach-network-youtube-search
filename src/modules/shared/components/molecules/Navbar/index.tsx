import React from "react";
import type { FC, ReactElement } from "react";
import Logo from "../../atoms/Logo";
import SearchForm from "../SearchForm";
import { StyledNav } from "./styles";

const Navbar: FC = (): ReactElement => {
  return (
    <StyledNav>
      <div className="nav-container">
        <Logo />
        <SearchForm />
      </div>
    </StyledNav>
  );
};

export default Navbar;
