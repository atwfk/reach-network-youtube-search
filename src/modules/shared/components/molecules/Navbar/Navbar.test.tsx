import renderer from "react-test-renderer";
import "jest-styled-components";
import Navbar from ".";

describe("Navbar Component snapshots", () => {
  it("Should render Navbar component", () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
