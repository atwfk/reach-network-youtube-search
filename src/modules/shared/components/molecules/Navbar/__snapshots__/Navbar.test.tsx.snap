// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component snapshots Should render Navbar component 1`] = `
.c1 {
  position: relative;
  margin-right: 10px;
}

.c1 span {
  position: absolute;
  display: none;
  top: 0;
  right: 20px;
  color: #9d9d9d;
  font-size: 12px;
}

.c1 .primary-logo {
  width: 180px;
  height: 40px;
  display: none;
}

.c1 .secondary-logo {
  width: 60px;
  height: 40px;
  background-color: red;
  display: block;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c3 input {
  width: 100%;
  padding: 5px;
  border: 1px solid #9d9d9d;
  font-size: 16px;
  outline: none;
  border-radius: 2px;
}

.c3 input:focus {
  border-color: #3ea6ff;
}

.c3 button {
  position: absolute;
  right: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c3 .remove-text {
  color: #9d9d9d;
}

.c4 {
  border: none;
  border-left: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
}

.c4 .search-icon {
  color: #eee;
  font-size: 20px;
  opacity: 0.6;
  margin-left: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.c0 {
  background-color: red;
  box-shadow: 0px 2px 2px rgba(0,0,0,0.2);
  padding: 5px 20px;
}

.c0 .nav-container {
  max-width: 920px;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media (min-width:768px) {
  .c1 {
    margin-right: 60px;
  }
}

@media (min-width:768px) {
  .c1 span {
    display: block;
  }
}

@media (min-width:768px) {
  .c1 .primary-logo {
    display: block;
  }
}

@media (min-width:768px) {
  .c1 .secondary-logo {
    display: none;
  }
}

@media (min-width:768px) {
  .c3 input {
    padding: 10px;
  }
}

@media (min-width:768px) {
  .c3 button {
    right: 20px;
  }
}

@media (min-width:768px) {
  .c4 {
    border: 1px solid #9d9d9d;
    padding: 0px 25px;
    opacity: 0.6;
    background-color: #eee;
  }
}

@media (min-width:768px) {
  .c4 .search-icon {
    color: #9d9d9d;
    margin: 0;
  }
}

@media (min-width:768px) {
  .c0 {
    background-color: white;
    padding: 10px 20px;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 2;
  }
}

<nav
  className="c0"
>
  <div
    className="nav-container"
  >
    <div
      className="c1"
    >
      <span>
        EG
      </span>
      <svg
        className="primary-logo"
      >
        primary-logo.svg
      </svg>
      <svg
        className="secondary-logo"
      >
        secondary-logo.svg
      </svg>
    </div>
    <form
      className="c2"
      onSubmit={[Function]}
    >
      <div
        className="c3"
      >
        <input
          data-testid="text-input"
          id="id_1"
          name="search"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
        
      </div>
      <button
        className="c4"
        data-testid="search-btn"
        disabled={true}
        type="submit"
      >
        <svg
          className="search-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
          />
        </svg>
      </button>
    </form>
  </div>
</nav>
`;
